Developer

dzmitry-levankou opened the pull request:
Check it out

dzmitry-levankou committed:
task #136 from leetcode

dzmitry-levankou committed:
task # 125 valid palindrome from leetcode

dzmitry-levankou committed:
task # 929 Unique Email Addresses from leetcode

sadfreddy commented on isPalindrome:
```diff
@@ -0,0 +1,13 @@
+        for i in s:
+            if i.isalpha() or i.isnumeric():
+                a += i.lower()
+        if a == a[::-1]:
```
return a == a[::-1]

sadfreddy replied:
это тоже надо поправить

dzmitry-levankou committed:
Practice replace() function

dzmitry-levankou committed:
added two more functions for practice

dzmitry-levankou committed:
Update replacePractice

The search function only works with entire words and this word must be in the list. Trim just removes all blanks from the list.

sadfreddy commented on replacePractice:
```diff
@@ -0,0 +1,33 @@
+s = ["a", "b", "c", "d"]
+
+# The number of words in the list will be returned if the list does not contain the whole word we are searching for.
+def findPractice(string, x):
```
назови более адекватно переменные, название переменных должно содержать их смысл (и сделай еще find для строк). findString("Hello", "el") -> 2 - это пример

sadfreddy commented on replacePractice:
```diff
@@ -0,0 +1,33 @@
+x = "bob"
+string = ["pope", "rome", "bob", "cat"]
+
+def trimPractice(b):
```
у тебя полностью пробелы убирает, а надо только лишние ("hello world" не должно становиться "helloworld")

sadfreddy commented on replacePractice:
```diff
@@ -0,0 +1,33 @@
+def replacePractice(s, l):
+    q = []
```
на вход должна приходить строка, а не лист (функция должна работать также как и тут https://www.w3schools.com/python/ref_string_replace.asp)

dzmitry-levankou committed:
Update replacePractice

A new findString function was added, as well as three other functions revised based on feedback provided

sadfreddy commented on replacePractice:
```diff
@@ -0,0 +1,43 @@
+            name = willBeReplacedByThat
+        emptyString += name
+    return emptyString
+haveToBeReplaced = "i"
```
а попробуй заменить строку, которая состоит из нескольких символов, например  replacePractice(hello, ell, l) -> hio

sadfreddy replied:
Строка, которая должна заменять другую строку, тоже должна передаваться в качестве параметра

sadfreddy replied:
Создай тест, который будет генерировать рандомные строки разных размеров и передавать их в существующую уже функцию https://www.w3schools.com/python/ref_string_replace.asp и в твою реализацию ( ответы везде должны быть одинаковыми)

sadfreddy commented on replacePractice:
```diff
@@ -0,0 +1,43 @@
+    newString = ""
+    for i in initialStr[1:-1]:
+        newString += i
+    return newString
```
initialStr = " bob junior    bobovich " на вот этом примере должен быть такой ответ "bob junior bobovich"

sadfreddy commented on replacePractice:
```diff
@@ -0,0 +1,43 @@
+stringList = ["pope", "rome", "bob", "cat"]
+print(findPractice(stringList, needsToBeFind))
+
+def findString(fromSearch):
```
что это функция должна делать?

dzmitry-levankou committed:
Update replacePractice

Symbols are now being replaced in the string by the function.

dzmitry-levankou committed:
Update replacePractice

The input is initialStr = "  bob junior bobovich  " (two spaces on the beginning and the end), the outcome is "bob junior bobovich" (without these mentioned spaces)

dzmitry-levankou committed:
Update isPalindrome

isPalindrome has been modified as requested

dzmitry-levankou committed:
Update replacePractice

replacePractice has been modified as requested

dzmitry-levankou committed:
Update replacePractice

I have modified and explained findString(fromSearch) method

dzmitry-levankou committed:
Create CheckSentence
